spopt/locate/flow.py

```diff
@@ -0,0 +1,2579 @@
+
+        weight : float, default=0.99
+            Controls the trade-off between node coverage and flow coverage for threshold extension objective.
+            Range: [0.0, 1.0]
+            - 1.0: Prioritises node coverage
+            - 0.0: Prioritises flow coverage
+            Only used when threshold > 0.
+        
+        objective : str, default="flow"
+            Optimisation objective.
+            - "flow": Maximize total flow coverage
+            - "vmt": Maximize vehicle miles traveled (VMT) coverage
+        
+        include_destination : bool, default=False
+            Determines how node weights are calculated in threshold extension.
+            
+            - False: Only origin nodes contribute to weight calculation
+            - True: Both origin and destination nodes contribute to weight calculation
+    """
+    def __init__(
+        self,
+        vehicle_range: Union[float, int] = 200000,
+        p_facilities: int = 5,
+        capacity: Union[float, None] = None,
+        threshold: float = 0.0,
+        weight: float = 0.99,
+        objective: str = "flow",
+        include_destination: bool = False
+    ):
+        self.vehicle_range = vehicle_range
+        self._original_vehicle_range = vehicle_range
+        self.p_facilities = p_facilities
+
+        self.capacity = capacity
+
+        self.threshold = threshold
+        self.weight = weight
+
+        self.objective = objective
+        self.include_destination = include_destination
+
+        self.network = None
+        self.flows = {}
+        self.flow_paths = {}
+        self.candidate_sites = []
+
+        self.facility_combinations = []
+        self.path_refueling_combinations = {}
+        self.e_coefficients = {}
+        self.scipy_network = None
+
+        self.K = {}  # K[path_id, arc_id] = [refueling_nodes]
+        self.a = {}  # a[path_id] = [arc_ids]
+
+        self.node_weights = None
+
+        self.selected_facilities = {}
+        self.flow_coverage = {}
+        self.covered_nodes = []
+        self.objective_value = 0
+        self.solution_time = 0
+        self.status = None
+
+        self.solver_type = None
+        self.variables = {}  
+        self.constraints = {}
+
+        self.model = None
+        self.lagrange_multipliers = {}  
+        self.reduced_costs = {} 
+        self.shadow_prices = {}
+
+        self.greedy_iterations = []
+        self.marginal_contributions = {}
+
+        self.solver_stats = {}
+
+        self.bounds = {"lower": None, "upper": None}
+
+        
+    def get_variable_value(self, var_name: str) -> Optional[float]:
+        return self.variables.get(var_name)
+
+    def get_constraint_dual(self, constraint_name: str) -> Optional[float]:
+        return self.lagrange_multipliers.get(constraint_name)
+
+    def get_reduced_cost(self, var_name: str) -> Optional[float]:
+        return self.reduced_costs.get(var_name)
+    
+    @classmethod
+    def convert_to_scipy_sparse(cls, network):
```

Comment:
Is this a classmethod? A classmethod should return a `FRLM` instance, not different instances of a graph. I think it is better to include this in `util.py`